pydantic_core/core_schema.py

```diff
@@ -304,54 +285,52 @@ def __call__(self, __input_value: Any, __index_key: int | str | None = None) ->
         ...
 
 
-class GeneralWrapSerializerFunction(Protocol):  # pragma: no cover
-    def __call__(
-        self, __input_value: Any, __serializer: SerializerFunctionWrapHandler, __info: SerializationInfo
-    ) -> Any:
-        ...
-
-
-class FieldWrapSerializerFunction(Protocol):  # pragma: no cover
-    def __call__(
-        self,
-        __model: Any,
-        __input_value: Any,
-        __serializer: SerializerFunctionWrapHandler,
-        __info: FieldSerializationInfo,
-    ) -> Any:
-        ...
-
+# (__input_value: Any, __serializer: SerializerFunctionWrapHandler) -> Any
+GeneralWrapNoInfoSerializerFunction = Callable[[Any, SerializerFunctionWrapHandler], Any]
+# (__input_value: Any, __serializer: SerializerFunctionWrapHandler, __info: SerializationInfo) -> Any
+GeneralWrapInfoSerializerFunction = Callable[[Any, SerializerFunctionWrapHandler, SerializationInfo], Any]
+# (__model: Any, __input_value: Any, __serializer: SerializerFunctionWrapHandler) -> Any
+FieldWrapNoInfoSerializerFunction = Callable[[Any, Any, SerializerFunctionWrapHandler], Any]
+# (__model: Any, __input_value: Any, __serializer: SerializerFunctionWrapHandler, __info: FieldSerializationInfo) -> Any
+FieldWrapInfoSerializerFunction = Callable[[Any, Any, SerializerFunctionWrapHandler, FieldSerializationInfo], Any]
+WrapSerializerFunction = Union[
+    GeneralWrapNoInfoSerializerFunction,
+    GeneralWrapInfoSerializerFunction,
+    FieldWrapNoInfoSerializerFunction,
+    FieldWrapInfoSerializerFunction,
+]
 
-class GeneralWrapSerializerFunctionSchema(TypedDict):
-    type: Literal['general']
-    function: GeneralWrapSerializerFunction
 
-
-class FieldWrapSerializerFunctionSchema(TypedDict):
-    type: Literal['field']
-    function: FieldWrapSerializerFunction
+class WrapSerializerFunctionSchema(TypedDict):
+    type: Literal['general', 'field']
+    info_arg: bool
+    function: WrapSerializerFunction
 
 
 class WrapSerializerFunctionSerSchema(TypedDict, total=False):
     type: Required[Literal['function-wrap']]
-    function: Required[Union[GeneralWrapSerializerFunctionSchema, FieldWrapSerializerFunctionSchema]]
-    schema: CoreSchema  # if ommited, the schema on which this serializer is defined is used
+    function: Required[WrapSerializerFunctionSchema]
+    schema: CoreSchema  # if omitted, the schema on which this serializer is defined is used
     json_return_type: JsonReturnTypes
     when_used: WhenUsed  # default: 'always'
```

Comment:
If we're going to merge `GeneralWrapSerializerFunctionSchema` and `FieldWrapSerializerFunctionSchema` why not just have all of their fields in `WrapSerializerFunctionSerSchema` directly?

Reply:
e.g. flatten to remove the `{'type': 'general', 'info_arg': True, 'function': f}` dict into the parent?

I guess we could, main pain point would be updating the tests.

Reply:
yeah that's what I meant. If we're going to get looser with the typing, at least let's save ourselves LOC / complexity.

Reply:
agreed.